api/migrations/20201002173359-addBcmiFlavourToMasterRecords.js

```diff
@@ -0,0 +1,178 @@
+      _schemaName: {
+        $in: [
+          'AnnualReportBCMI',
+          'CertificateAmendmentBCMI',
+          'CorrespondenceBCMI',
+          'DamSafetyInspectionBCMI',
+          'InspectionBCMI',
+          'ManagementPlanBCMI',
+          'OrderBCMI',
+          'PermitBCMI',
+          'ReportBCMI'
+        ]
+      }
+    }).toArray();
+    let flavourMasterIds = existingBcmiFlavours.map(flavour => flavour._master);
+
+    let masterRecords = await nrpti.find({
+      _schemaName: {
+        $in: [
+          'AnnualReport',
+          'CertificateAmendment',
+          'Correspondence',
+          'DamSafetyInspection',
+          'Inspection',
+          'ManagementPlan',
+          'Order',
+          'Permit',
+          'Report'
+        ]
+      }
+    }).toArray();
+
+    let promises = [];
+    let i = masterRecords.length;
+
+    while (i--) {
+      const record = masterRecords[i];
+      let j = flavourMasterIds.length
+      let hasBcmiFlavour = false;
+      while (j--) {
+        const flavourMasterId = flavourMasterIds[j];
+        if (record._id.toString() === flavourMasterId.toString()) {
+          flavourMasterIds.splice(j, 1);
+          hasBcmiFlavour = true;
+          break;
+        }
+      }
+
+      if (!hasBcmiFlavour) {
+        let args = {
+          swagger: {
+            params: {
+              auth_payload: {
+                realm_access: {
+                  roles: ['sysadmin']
+                }
+              }
+            }
+          }
+        };
+
+        let flavourRecord = {
+          ...record
+        };
+        let obj = null;
+
+        flavourRecord._master = record._id;
+
+        switch (record._schemaName) {
+          case 'AnnualReport':
+            obj = AnnualReport.createBCMI(args, null, null, flavourRecord);
+            break;
+          case 'CertificateAmendment':
+            obj = CertificateAmendment.createBCMI(args, null, null, flavourRecord);
+            break;
+          case 'Correspondence':
+            obj = Correspondence.createBCMI(args, null, null, flavourRecord);
+            break;
+          case 'DamSafetyInspection':
+            obj = DamSafetyInspection.createBCMI(args, null, null, flavourRecord);
+            break;
+          case 'Inspection':
+            obj = Inspection.createBCMI(args, null, null, flavourRecord);
+            break;
+          case 'ManagementPlan':
+            obj = ManagementPlan.createBCMI(args, null, null, flavourRecord);
+            break;
+          case 'Order':
+            obj = Order.createBCMI(args, null, null, flavourRecord);
+            break;
+          case 'Permit':
+            obj = Permit.createBCMI(args, null, null, flavourRecord);
+            break;
+          case 'Report':
+            obj = Report.createBCMI(args, null, null, flavourRecord);
+            break;
+          default:
+            break;
+        }
+
+        if (obj) {
+          promises.push(await obj.save());
```

Comment:
I don't think the `await` should be here as it is awaiting the promise result before pushing it onto the array. They should all resolve in line 163 instead.

Reply:
5d4da993f36eca4bdc1a327ab7784901d7204cc3